### ISSUE TITLE
onTrackUpdate not called when local peer joins the room

### ISSUE THREAD

**Issue by abbas2501** [100mslive/100ms-flutter#297]
In a previous version of the SDK, it was being called. However, in the latest update - it seems the opposite.

Also - when I go online from the web panel (HMS dashboard) and join the same room, there it shows my flutter app's joined user in the user's list but without video/audio. 
I do have necessary permissions in Manifest.

Is it something on my end or the sdk itself ? 

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-arm, locale
    en-IN)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] VS Code (version 1.62.0)
[✓] Connected device (2 available)
```

**Comment by abbas2501** [comment]
hmssdk_flutter: ^0.4.0


```
def localProperties = new Properties()
def localPropertiesFile = rootProject.file('local.properties')
if (localPropertiesFile.exists()) {
    localPropertiesFile.withReader('UTF-8') { reader ->
        localProperties.load(reader)
    }
}

def flutterRoot = localProperties.getProperty('flutter.sdk')
if (flutterRoot == null) {
    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
}

def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
if (flutterVersionCode == null) {
    flutterVersionCode = '1'
}

def flutterVersionName = localProperties.getProperty('flutter.versionName')
if (flutterVersionName == null) {
    flutterVersionName = '1.0'
}

apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
apply plugin: 'com.google.gms.google-services'
apply plugin: 'com.google.firebase.crashlytics'
apply plugin: 'com.google.firebase.firebase-perf'


 def keystoreProperties = new Properties()
   def keystorePropertiesFile = rootProject.file('key.properties')
   if (keystorePropertiesFile.exists()) {
       keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
   }


android {
    compileSdkVersion 31

    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }

    lintOptions {
        disable 'InvalidPackage'
    }

    defaultConfig {
        applicationId "xyz"
        minSdkVersion 21
        targetSdkVersion 30
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
        multiDexEnabled true
    }
    ndkVersion '22.0.6917172 rc1'


/// Old Added by [Bhargav]

//    signingConfigs{
//        debug{
//            keyAlias 'key0'
//            keyPassword 'Zuzu@2020'
//            storeFile file('/Users/blb/Documents/Zuzu_customer')
//            storePassword 'Zuzu@2020'
//        }
//    }

//     buildTypes {
//         debug{
//             signingConfig signingConfigs.debug
//         }
//         release {
//             shrinkResources false
//             minifyEnabled false
//             // Signing with the debug keys for now, so `flutter run --release` works.
//             signingConfig signingConfigs.debug
// //            ndk {
// //                debugSymbolLevel 'SYMBOL_TABLE' | 'FULL'
// //            }

//         }
//     }

/// New Added by [Abbas]
 signingConfigs {
       release {
           keyAlias keystoreProperties['keyAlias']
           keyPassword keystoreProperties['keyPassword']
           storeFile keystoreProperties['storeFile'] ? file(keystoreProperties['storeFile']) : null
           storePassword keystoreProperties['storePassword']
       }
   }
   buildTypes {
       release {
           signingConfig signingConfigs.release
       }
   }
   
    buildToolsVersion '30.0.3'
    compileOptions {
        sourceCompatibility "1.8"
        targetCompatibility "1.8"
    }
    dependenciesInfo {
        includeInApk true
        includeInBundle true
    }
//    flavorDimensions "app"
//
//    productFlavors {
//        dev {
//            dimension "app"
//            resValue "string", "app_label", "Zuzu-Dev"
//            versionNameSuffix "-dev"
//        }
//        prod {
//            dimension "app"
//            resValue "string", "app_label", "Zuzu"
//        }
//    }
}
flutter {
    source '../..'
}

dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation 'com.google.firebase:firebase-analytics:19.0.0'
    implementation 'com.google.firebase:firebase-messaging:22.0.0'
    implementation 'androidx.multidex:multidex:2.0.1'
    implementation 'com.google.firebase:firebase-crashlytics:18.0.0'

    androidTestImplementation 'androidx.test:runner:1.3.0'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'

    implementation 'com.facebook.android:facebook-android-sdk:5.15.3'

    //App Check sefteyNet
    implementation "androidx.browser:browser:1.3.0"

    // 100ms
    implementation 'com.github.100mslive:android-sdk:2.1.8'

}

```

**Comment by abbas2501** [comment]
hmssdk_flutter: ^0.4.0


```
def localProperties = new Properties()
def localPropertiesFile = rootProject.file('local.properties')
if (localPropertiesFile.exists()) {
    localPropertiesFile.withReader('UTF-8') { reader ->
        localProperties.load(reader)
    }
}

def flutterRoot = localProperties.getProperty('flutter.sdk')
if (flutterRoot == null) {
    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
}

def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
if (flutterVersionCode == null) {
    flutterVersionCode = '1'
}

def flutterVersionName = localProperties.getProperty('flutter.versionName')
if (flutterVersionName == null) {
    flutterVersionName = '1.0'
}

apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
apply plugin: 'com.google.gms.google-services'
apply plugin: 'com.google.firebase.crashlytics'
apply plugin: 'com.google.firebase.firebase-perf'


 def keystoreProperties = new Properties()
   def keystorePropertiesFile = rootProject.file('key.properties')
   if (keystorePropertiesFile.exists()) {
       keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
   }


android {
    compileSdkVersion 31

    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }

    lintOptions {
        disable 'InvalidPackage'
    }

    defaultConfig {
        applicationId "xyz"
        minSdkVersion 21
        targetSdkVersion 30
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
        multiDexEnabled true
    }
    ndkVersion '22.0.6917172 rc1'

   
    buildToolsVersion '30.0.3'
    compileOptions {
        sourceCompatibility "1.8"
        targetCompatibility "1.8"
    }
    dependenciesInfo {
        includeInApk true
        includeInBundle true
    }

flutter {
    source '../..'
}

dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation 'com.google.firebase:firebase-analytics:19.0.0'
    implementation 'com.google.firebase:firebase-messaging:22.0.0'
    implementation 'androidx.multidex:multidex:2.0.1'
    implementation 'com.google.firebase:firebase-crashlytics:18.0.0'

    androidTestImplementation 'androidx.test:runner:1.3.0'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'

    implementation 'com.facebook.android:facebook-android-sdk:5.15.3'

    //App Check sefteyNet
    implementation "androidx.browser:browser:1.3.0"

    // 100ms
    implementation 'com.github.100mslive:android-sdk:2.1.8'

}
```

**Comment by ygit** [comment]
@abbas2501 This issue will be resolved in the coming release of the Flutter SDK

**Comment by abbas2501** [comment]
Is this solved ?

**Comment by ygit** [comment]
Resolved in version 0.5.0
